src/eduid_webapp/authn/views.py

```diff
@@ -259,23 +259,66 @@ def logout_service():
+def signup_authn():
+    current_app.logger.debug('Authenticating signing up user')
+
+    # XXX TRANSITION_TOKEN_LOGIN remove after transition to implicit logins
+    location_on_fail = current_app.config.get('TOKEN_LOGIN_FAILURE_REDIRECT_URL')
+    location_on_success = current_app.config.get('TOKEN_LOGIN_SUCCESS_REDIRECT_URL')
+    token = request.form.get('token', None)
+    if token is not None:
+        eppn = request.form.get('eppn')
+        nonce = request.form.get('nonce')
+        timestamp = request.form.get('ts')
+        loa = get_loa(current_app.config.get('AVAILABLE_LOA'), None)  # With no session_info lowest loa will be returned
+        shared_key = current_app.config.get('SIGNUP_AND_AUTHN_SHARED_KEY')
+
+        if verify_auth_token(shared_key=shared_key, eppn=eppn, token=token, nonce=nonce, timestamp=timestamp,
+                             usage='signup_login'):
+            try:
+                user = current_app.central_userdb.get_user_by_eppn(eppn)
+                if user.locked_identity.count > 0:
+                    # This user has previously verified their account and is not new, this should not happen.
+                    current_app.logger.error('Not new user {} tried to log in using token login'.format(user))
+                    return redirect(location_on_fail)
+                session['eduPersonPrincipalName'] = user.eppn
+                session['user_eppn'] = user.eppn
+                session['user_is_logged_in'] = True
+                session['eduPersonAssurance'] = loa
+                response = redirect(location_on_success)
+                #session.set_cookie(response)  # Is this needed as session_interface.save_session will be called?
+                current_app.logger.info('Successful token login, redirecting user {} to {}'.format(user,
+                                                                                                   location_on_success))
+                return response
+            except current_app.central_userdb.exceptions.UserDoesNotExist:
+                current_app.logger.error('No user with eduPersonPrincipalName = {} found'.format(eppn))
+            except current_app.central_userdb.exceptions.MultipleUsersReturned:
+                current_app.logger.error("There are more than one user with eduPersonPrincipalName = {}".format(eppn))
+
+        current_app.logger.info('Token login failed, redirecting user to {}'.format(location_on_fail))
+        return redirect(location_on_fail)
+    # XXX END remove
+
+    location_on_fail = current_app.config.get('SIGNUP_AUTHN_FAILURE_REDIRECT_URL')
+    location_on_success = current_app.config.get('SIGNUP_AUTHN_SUCCESS_REDIRECT_URL')
+
+    eppn = check_previous_identification(session.signup)
+    if eppn is not None:
+        loa = get_loa(current_app.config.get('AVAILABLE_LOA'), None)  # With no session_info lowest loa will be returned
+        current_app.logger.info("Starting pre-login actions " "for eppn: {})".format(eppn))
```

Comment:
What does "Starting pre-login actions" mean in  this context? Copy paste error?

Reply:
Exactly, sorry.